pyaedt/modeler/calculators.py

```diff
@@ -0,0 +1,400 @@
+
+        k0 = 2.0 * math.pi * f / 3.0e8
+
+        ustrip_length = math.radians(electrical_length) / (math.sqrt(er_eff) * k0)
+
+        return ustrip_width, ustrip_length
+
+    @pyaedt_function_handler()
+    def microstrip_analysis(self, substrate_height, permittivity, width, thickness):
+        """Strip line calculator.
+
+        Parameters
+        ----------
+        substrate_height : float
+            Substrate height.
+        permittivity : float
+            Substrate permittivity.
+        width : float
+            Trace width.
+        thickness : float
+            Trace thickness.
+
+        Returns
+        -------
+        float
+            z0
+        """
+        e0 = permittivity
+        h0 = substrate_height
+        z1 = 87 / (math.sqrt(e0 + 1.41))
+        z2 = math.log(5.98 * h0 / ((0.8 * width) + thickness))
+        z0 = z1 * z2
+        return z0
+
+    @pyaedt_function_handler()
+    def differential_microstrip_analysis(self, substrate_height, permittivity, width, separation, thickness):
+        """Strip line calculator.
+
+        Parameters
+        ----------
+        substrate_height : float
+            Substrate height.
+        permittivity : float
+            Substrate permittivity.
+        width : float
+            Trace width.
+        separation : float
+            Trace separation.
+        thickness : float
+            Trace thickness.
+
+
+        Returns
+        -------
+        tuple
+            z0 single end and differential.
+        """
+        e0 = permittivity
+        h0 = substrate_height
+        z1 = 87 / (math.sqrt(e0 + 1.41))
+        z2 = math.log(5.98 * substrate_height / ((0.8 * width) + thickness))
+        z0 = z1 * z2
+        z0d = 2 * z0 * (1 - (0.48 * math.exp(-0.96 * separation / h0)))
+        return z0, z0d
+
+    @pyaedt_function_handler()
+    def stripline_synthesis(self, substrate_height, permittivity, impedance=50.0):
+        """Strip line calculator.
+
+        Parameters
+        ----------
+        substrate_height : float
+            Substrate height.
+        permittivity : float
+            Substrate permittivity.
+        impedance : str, optional
+            Impedance. The default is ``50.0``.
+
+        Returns
+        -------
+        float
+            Line width.
+        """
+
+        x = 30.0 * math.pi / (math.sqrt(permittivity) * impedance) - 0.441
+
+        if math.sqrt(permittivity) * impedance <= 120:
+            w_over_h = x
+        else:
+            w_over_h = 0.85 - math.sqrt(0.6 - x)
+
+        width = w_over_h * substrate_height
+        return width
+
+    @pyaedt_function_handler()
+    def suspended_strip_synthesis(self, substrate_height, permittivity, w1, units="mm"):
+        """Suspended stripline calculator.
+
+        Parameters
+        ----------
+        substrate_height : float
+            Substrate in meter.
+        permittivity : float
+            Dielectric permittivity
+        w1 : float
```

Comment:
Add a description for w1 and an entry for units?

Reply:
I'll add in next PR since this is a first class definition. Thanks for notifying